app/js/MongoManager.js

```diff
@@ -44,15 +44,17 @@ module.exports = MongoManager = {
     if (!options.include_deleted) {
       query.deleted = { $ne: true }
     }
-    return db.docs.find(query, filter, callback)
+    docsCollectionPromise.then((docs) =>
+      docs.find(query, filter).toArray(callback)
```

Comment:
Just noting that at some point it might be better to use the cursor directly instead of calling `toArray` as this could be expensive for a big project. - I realise that this is what `find` does in `mongojs` but perhaps something we could clean up in future.

Reply:
https://github.com/overleaf/issues/issues/3458